pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -553,6 +547,26 @@ private void createPersistentSubscriptions() {
         checkReplicatedSubscriptionControllerState();
     }
 
+    private CompletableFuture<Void> removeOrphanReplicationCursors() {
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        String localCluster = brokerService.pulsar().getConfiguration().getClusterName();
+        List<String> replicationClusters = topicPolicies.getReplicationClusters().get();
+        for (ManagedCursor cursor : ledger.getCursors()) {
+            if (cursor.getName().startsWith(replicatorPrefix)) {
+                String remoteCluster = PersistentReplicator.getRemoteCluster(cursor.getName());
+                if (!replicationClusters.contains(remoteCluster)) {
+                    log.warn("Remove the replicator because the cluster '{}' does not exist", remoteCluster);
+                    futures.add(removeReplicator(remoteCluster));
+                    continue;
+                }
+                if (localCluster.equals(remoteCluster)) {
+                    log.warn("Remove the replicator because the cluster '{}' does not exist", remoteCluster);
+                }
```

Comment:
Does something get removed in this case? Please add a unique log message for this case which explains the situation.

Reply:
> Does something get removed in this case? 

No, removed